eng/scripts/smoketest/Get-Package-Version.ps1

```diff
@@ -0,0 +1,73 @@
+. (Join-Path $PSScriptRoot ../../common/scripts/common.ps1)
+$ErrorActionPreference = 'Continue'
+
+$NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v2"
+$allPackages = Get-AllPkgProperties
+$trackTwoPackages = $allPackages.Where({ $_.IsNewSdk })
+
+$azureCorePkgInfo = $trackTwoPackages.Where({ $_.Name -eq "Azure.Core"})
+$azureCoreVer = [AzureEngSemanticVersion]::ParseVersionString($azureCorePkgInfo.Version)
+$azureCoreVer.IsPreRelease = $false
+$azureCoreVerBase = $azureCoreVer.ToString()
+
+$currentDate = Get-Date -Format "yyyyMM"
+$currentDate = [DateTime]::ParseExact($currentDate, "yyyyMM", $null)
+$azureCoreMinDate = $currentDate.addMonths(-6).ToString("yyyyMM")\
+
+$azureCorePkg = Find-Package -Name 'Azure.Core' `
```

Comment:
I don't think this covers all of the scenarios that can cause cross-threaded versioning issues.   Any projects that contain project references to another library in the repository would suffer from the same invalid version.   For example,  `Azure.Messaging.EventHubs.Processor` using a project reference to `Azure.Messaging.EventHubs`  or  `Azure.Data.SchemaRegistry` using project references to `Azure.Core.Experimental` and to `Azure.Messaging.EventHubs`.   

I believe that we'll need to do this for all packages.

Reply:
@jsquire I agree with you that this doesn't cover every scenario but I believe it does handle most of the scenarios and if needed we can harden it further by adding more levels of versioning checks with more than just Core.

One thing to keep in mind is that any Project dependencies within a given service directory (i.e. Azure.Messaging.EventHubs.Process and Azure.Messaging.EventHubs) shouldn't ever be a real issue because they build and publish together. If one of those builds fails than neither of the daily packages will publish so they should never be out of sync. The only time project references might cause things to be out of sync is when they are across service directory boundaries. Azure.Core and Azure.Core.Experimental are probably the two most common cross service directory project references and they publish together from the core service directory so they should also always be in sync with each other as well. So, using Azure.Core as the base dependency I believe should make most scenarios work and if needed we can make adjustments if we find otherwise.

Reply:
@weshaggard:  My fear is that we'll see the same pattern of intermittent failures that we have in the past, and the smoke tests will start being ignored and neglected again.    While I agree that Core is the 80% case, I don't think other scenarios are rare enough that we won't see problems at least a couple of time a month if we restrict this to Core only.

Would it be possible for us to walk the dev feed and validate that all dependency references for a package exist?   If they do not, we step back to the previous version and validate that?  We use the one that is healthy for the smoke tests?   Unless I'm not thinking of it correctly, I believe that would work and we'd have a worst case where the smoke tests were validating some packages a day late.  If both current and previous are broken, we can blow up and raise alarms that there's an issue with the references and someone should investigate.

Thoughts?

Reply:
Yes I think it would be possible to do a full dependency walk but it would be a lot more complex, and potentially error prone and that is why I think starting with Core is a simpler strategy. If that turns out to still cause issues then we can continue to work on improving the strategy here.

Another thing I worry about if we blindly use the dependency walk is that some packages that might have been failing for a while may end up in pretty stale state, as in we haven't validated them (and any connected packages) for weeks or more and we wouldn't have any indication that they no longer work. I don't want one bad apple to ruin all the other tests. 

Perhaps another middle ground is to add some additional logging here for any cases where we find package versions that are older than the core reference version as those could potentially cause errors. That would at least give us some help when diagnosing any potential failures.

I will also add that I think a lot of the troubles with the older smoke-test pipeline was more around the tests we were running and hopefully that is less of an issue with the new approach.

Reply:
We definitely need to include all `Azure.Core.*` packages for the special logic.    I know of at least 3 different project references to `Azure.Core.Experimental` at the moment.

Reply:
Should we also include the track 2s that are still Microsoft.* The ones I know about are all in sdk/core and sdk/extensions. This was done to avoid confusion that these were client libraries, but they are still utilities for working with T2 libraries.